backend/moonstreamapi/routes/queries.py

```diff
@@ -0,0 +1,384 @@
+        logger.error(f"Error in updating query: {e}")
+        raise MoonstreamHTTPException(status_code=500, internal_error=e)
+
+    return entry
+
+
+@router.put("/{query_name}", tags=["queries"])
+async def update_query_handler(
+    request: Request,
+    query_name: str,
+    request_update: data.UpdateQueryRequest = Body(...),
+) -> BugoutJournalEntryContent:
+
+    token = request.state.token
+
+    query_id = get_query_by_name(query_name, token)
+
+    try:
+
+        entry = bc.update_entry_content(
+            token=MOONSTREAM_ADMIN_ACCESS_TOKEN,
+            journal_id=MOONSTREAM_QUERIES_JOURNAL_ID,
+            entry_id=query_id,
+            title=query_name,
+            content=request_update.query,
+            tags=["preapprove"],
+        )
+
+    except BugoutResponseException as e:
+        logger.error(f"Error in updating query: {str(e)}")
+        raise MoonstreamHTTPException(status_code=e.status_code, detail=e.detail)
+
+    except Exception as e:
+        logger.error(f"Error in updating query: {e}")
+        raise MoonstreamHTTPException(status_code=500, internal_error=e)
+
+    return entry
+
+
+@router.post("/{query_name}/update_data", tags=["queries"])
+async def update_query_data_handler(
+    request: Request,
+    query_name: str,
+    request_update: data.UpdateDataRequest = Body(...),
+) -> Optional[Dict[str, Any]]:
+    """
+    Request update data on S3 bucket
+    """
+
+    token = request.state.token
+
+    query_id = get_query_by_name(query_name, token)
+
+    try:
+        entries = bc.search(
+            token=MOONSTREAM_ADMIN_ACCESS_TOKEN,
+            journal_id=MOONSTREAM_QUERIES_JOURNAL_ID,
+            query=f"#approved #query_id:{query_id} !#preapprove",
+            limit=1,
+            timeout=5,
+        )
+
+        if entries.results and entries.results[0].content:
+            content = entries.results[0].content
+
+            tags = entries.results[0].tags
+
+            file_type = "json"
+
+            if "ext:csv" in tags:
+                file_type = "csv"
+
+            responce = requests.post(
+                f"{MOONSTREAM_CRAWLERS_SERVER_URL}:{MOONSTREAM_CRAWLERS_SERVER_PORT}/jobs/{query_id}/query_update",
+                json={
+                    "query": content,
+                    "params": request_update.params,
+                    "file_type": file_type,
+                },
+                timeout=5,
+            )
+
+            if responce.status_code != 200:
+                raise MoonstreamHTTPException(
+                    status_code=responce.status_code,
+                    detail="Task for start generate stats failed.",
+                )
+
+            return responce.json()
+    except Exception as e:
+        logger.error(f"Error in send generate query data task: {e}")
+        raise MoonstreamHTTPException(status_code=500, internal_error=e)
+    raise MoonstreamHTTPException(status_code=403, detail="Query not approved yet.")

```

Comment:
Wrong structure, better to use if `len(entries) == 0` then raise `Query not approved yet.`
But end return shoud exists, otherwise, there no possibility to define response_model
```
@router.post("/{query_name}/update_data", tags=["queries"], response_model=SomeDataModel)
```